go-controller/pkg/util/nicstobridge_test.go

```diff
@@ -0,0 +1,1218 @@
+package util
+
+import (
+	"bytes"
+	"fmt"
+	"testing"
+
+	ovntest "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/testing"
+	netlink_mocks "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/testing/mocks/vishvananda/netlink"
+	"github.com/vishvananda/netlink"
+
+	mock_k8s_io_utils_exec "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/testing/mocks/k8s.io/utils/exec"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util/mocks"
+	"github.com/stretchr/testify/mock"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetNicName(t *testing.T) {
+	mockKexecIface := new(mock_k8s_io_utils_exec.Interface)
+	mockExecRunner := new(mocks.ExecRunner)
+	mockCmd := new(mock_k8s_io_utils_exec.Cmd)
+	// below is defined in ovs.go
+	runCmdExecRunner = mockExecRunner
+	// note runner is defined in ovs.go file
+	runner = &execHelper{exec: mockKexecIface}
+	tests := []struct {
+		desc                string
+		errExp              bool
+		outputEmpty         bool
+		inpBrName           string
+		onRetArgsExecIface  *onCallReturnArgs
+		onRetArgsKexecIface *onCallReturnArgs
+	}{
+		{
+			desc:                "empty string as inputBrName",
+			errExp:              true,
+			outputEmpty:         true,
+			inpBrName:           "",
+			onRetArgsExecIface:  &onCallReturnArgs{"RunCmd", []string{"*mocks.Cmd", "string", "[]string", "string", "string", "string", "string"}, []interface{}{bytes.NewBuffer([]byte("")), bytes.NewBuffer([]byte("")), fmt.Errorf("test error")}},
+			onRetArgsKexecIface: &onCallReturnArgs{"Command", []string{"string", "string", "string", "string", "string"}, []interface{}{mockCmd}},
+		},
+		{
+			desc:                "inpBrName WITHOUT prefix `br` and stdout empty",
+			outputEmpty:         true,
+			inpBrName:           "int",
+			onRetArgsExecIface:  &onCallReturnArgs{"RunCmd", []string{"*mocks.Cmd", "string", "[]string", "string", "string", "string", "string"}, []interface{}{bytes.NewBuffer([]byte("")), bytes.NewBuffer([]byte("")), nil}},
+			onRetArgsKexecIface: &onCallReturnArgs{"Command", []string{"string", "string", "string", "string", "string"}, []interface{}{mockCmd}},
+		},
+		{
+			desc:                "inpBrName WITH prefix `br` and stdout empty",
+			inpBrName:           "br-int",
```

Comment:
where in this test do you actually check that the return value is "-int"?

Reply:
The test is currently not comparing the actual output returned currently

Reply:
I have now fixed the test case to compare output expected